glommio/src/sys/hardware_topology.rs

```diff
@@ -93,6 +93,47 @@ pub fn get_machine_topology_unsorted() -> io::Result<Vec<CpuLocation>> {
         }
     }
 
+    // Make sure that differnt NUMA nodes do not contain the same core id.
+    let mut core_id_not_unique = false;
+    let mut core_to_numa = HashMap::new();
+    for l in &cpu_locations {
+        if *core_to_numa.entry(l.core).or_insert(l.numa_node) != l.numa_node {
+            core_id_not_unique = true;
+            break;
+        }
+    }
+
+    if core_id_not_unique {
```

Comment:
this would work as well if the core id is unique, right?
To keep things simpler and easier to reason, I'd just invoke this all the time so we have the same code paths in all cases.

Reply:
Yes, it works.

Reply:
Fixed in https://github.com/DataDog/glommio/pull/480/commits/4aac955cf464a4acee1703ea3ede2ff933f86397.